gsd/gsd.c

```diff
@@ -62,6 +62,46 @@ size_t pwrite(int fd, void *buf, size_t count, int64_t offset)
 
 #endif
 
+static ssize_t __pwrite_retry(int fd, const void *buf, size_t count, int64_t offset)
+    {
+    ssize_t total_bytes_written = 0;
+    const char *ptr = (char *)buf;
+
+    // perform multiple pwrite calls to complete a large write sucessfully
+    while (total_bytes_written < count)
+        {
+        ssize_t bytes_written = pwrite(fd, ptr + total_bytes_written, count - total_bytes_written, offset + total_bytes_written);
```

Comment:
It might be useful to check if bytes_written is 0 to prevent infinite loops.
That might be a scenario if the physical is exhausted. The man page, write(2) and pwrite(1) states, that pwrite can return less bytes written if the physical devices is exhausted.
Since it is unlikely that this will change between loop iterations an infinite loop is entered.

Reply:
The man page indicates that write will return -1 on errors like this, but it doesn't hurt to check for the case where `bytes_written` is 0 and `errno` is set.

Reply:
Maybe I misunderstood the man page. Sorry.

Reply:
Code review is appreciated and increases the quality of the code. I added the check for `bytes_written == 0` you suggested just with `&& errno != 0` to make sure that there was an error that caused this return value.